options.go

```diff
@@ -61,3 +72,19 @@ func WithUsageHeader(h string) Option {
 		o.withUsageHeader = h
 	}
 }
+
+// WithQuotaStorageCapacityMetric is used to provide a metric that will record
+// the total capacity available to the Limiter for storing Quotas.
+func WithQuotaStorageCapacityMetric(g metric.Gauge) Option {
+	return func(o *options) {
```

Comment:
should we consider checking for a nil interface and returning an error?  I know
that means changing the options pattern to include the possibility of returning
an error.

Reply:
perhaps, it's okay to have an opinionated implementation which doesn't allow for
such "not particularly good implementation" of a gauge?  It's not like the devs
won't be able to easily use the pkg even if it is opinionated.  There's a
benefit as well, that it catches the far more typical use-case and prevents a
possible runtime panic.

Reply:
According to [this](https://go.dev/tour/methods/12) (my emphasis):

> If the concrete value inside the interface itself is nil, the method will be called with a nil receiver.
>
> In some languages this would trigger a null pointer exception, but **in Go it is common to write methods that gracefully handle being called with a nil receiver**

You could argue that it actually isn't as common as go thinks it is, but I think I would rather align with the language on this. There might be some good implementations that also allow for a nil receiver, there might even be some that require it to be nil to function a certain way, in which case having the library be opinionated would prevent it from being used. 

I can update it so we call `Set` on both metrics within `newExpirableStore`. Currently it calls set on the capacity metric, but by calling Set on both, it would at least mean if there was a panic it would hopefully happen early on when a program is initializing things. This would allow it to be discovered and fixed by that program however it chooses to do so.

Reply:
All due respect to the authors of a "Tour of Go" but I think it's exceedingly rare to see receiver methods that gracefully handle being called with a nil receiver.

Reply:
>  I know that means changing the options pattern to include the possibility of returning an error.

Doing this makes using options very cumbersome to use. However, I could add a check when the options are used in `newExpirableStore` and return an error there if either gauge is a nil interface, however...

>  There's a benefit as well, that it ... prevents a possible runtime panic.

It seems like this is almost certainly a programing error, in which case a panic seems like the correct action to take, and since this panic would happen in the same place where it would otherwise make sense to return an error, why not just panic so the bug can be identified and fixed? And that still allows the possibility for an implementation of a gauge that is something like I mention above, even if it is rare.

Reply:
Calling `Set` on both would at least force a panic which is better than deferring it.